test/Integration/Porter/PorterTest.php

```diff
@@ -121,11 +121,38 @@ public function testImport()
 
         self::assertInstanceOf(PorterRecords::class, $records);
         self::assertNotSame($this->specification, $records->getSpecification());
+        self::assertInstanceOf(CountableProviderRecords::class, $records->getPreviousCollection());
+        self::assertSame('foo', $records->current());
+    }
+
+    public function testNonCountableIteratorImport()
+    {
+        $porter = (new Porter)->registerProvider(
+            $this->provider =
+                \Mockery::mock(Provider::class)
+                    ->shouldReceive('fetch')
+                    ->andReturn($this->iterateOne())
```

Comment:
Prefer specifying an anonymous callback instead of writing a private test method because test cases should be self-contained. This can be expressed using `andReturnUsing` as in the following example.

```php
->andReturnUsing(function () {
    yield 'foo';
})
```